### DISCUSSION TITLE
string.Empty as a Const

### DISCUSSION THREAD

**Discussion by eidylon** [dotnet/runtime#66300] [General] [answered]
I'm curious if there is a specific reason, or not, as to why `string.Empty` is a field, and not a const? 

It's always kind of bugged me that I when defining a function with optional parameters, I can't do 

```
function myFunc(string optparam = string.Empty) { ... }
```

as it then yells at me that "optional parameter defaults must be compile-time constants".

**Comment by CyrusNajmabadi** [comment]
Moving discussion.  Nothing in C# dictates this afaict. It's just a decision of a particular release of the runtime.

**Comment by teo-tsirpanis** [comment, marked as answer]
Duplicate of #60508. Quoting myself from there:

> I persume you want to add this constant field to completely avoid writing `""` in your code, based on the knowledge that magic strings are a bad practice. But the empty string is not a magic string per se, you could view its two consecutive double quotes as a _symbol_ like `||` or `&&`.

**Reply by eidylon** [reply]
Exactly. Just kinda don't like having hard-coded values in my code if I can help it. 
Just like using `Environment.NewLine` instead of `"\n"`.

**Reply by CyrusNajmabadi** [reply]
> Exactly. Just kinda don't like having hard-coded values in my code if I can help it.

You can make your own const value that is set to the empty string and then just refer to that wherever you want a name instead of a literal.
